server/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecMergeIT.java

```diff
@@ -0,0 +1,175 @@
+
+        Map<String, String> codecMap = Map.of(
+            "best_compression",
+            "BEST_COMPRESSION",
+            "zstd_no_dict",
+            "ZSTD_NO_DICT",
+            "zstd",
+            "ZSTD",
+            "default",
+            "BEST_SPEED"
+        );
+
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            forceMergeMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
+        }
+
+    }
+
+    private void forceMergeMultipleCodecs(String finalCodec, String finalCodecMode, Map<String, String> codecMap) throws ExecutionException,
+        InterruptedException {
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String index = "test-index" + finalCodec;
+
+        // creating index
+        createIndex(
+            index,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put("index.codec", "default")
+                .put("index.merge.policy.max_merged_segment", "1b")
+                .build()
+        );
+        ensureGreen(index);
+        // ingesting and asserting segment codec mode for all four codecs
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            useCodec(index, codec.getKey());
+            ingestUsingCodec(index);
+        }
+
+        assertTrue(
+            getSegments(index).stream()
+                .flatMap(s -> s.getAttributes().values().stream())
+                .collect(Collectors.toSet())
+                .containsAll(codecMap.values())
+        );
+
+        // force merge into final codec
+        useCodec(index, finalCodec);
+        ingestUsingCodec(index);
+        final ForceMergeResponse forceMergeResponse = client().admin().indices().prepareForceMerge(index).setMaxNumSegments(1).get();
+
+        assertThat(forceMergeResponse.getFailedShards(), is(0));
+        assertThat(forceMergeResponse.getSuccessfulShards(), is(2));
+
+        flushAndRefreshIndex(index);
+
+        List<Segment> segments = getSegments(index).stream().filter(Segment::isSearch).collect(Collectors.toList());
+        assertEquals(1, segments.size());
+        assertTrue(segments.stream().findFirst().get().attributes.containsValue(finalCodecMode));
+    }
+
+    private void useCodec(String index, String codec) throws ExecutionException, InterruptedException {
+        assertAcked(client().admin().indices().prepareClose(index));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index));
+    }
+
+    private void ingestUsingCodec(String index) throws InterruptedException {
```

Comment:
This doesn't do anything specific to codec, can be called ```ingestDocs``` or ```insertDocs```

Reply:
done